mindmeld/text_preparation/normalizers.py

```diff
@@ -195,28 +196,105 @@ def normalize(self, text):
         return unicodedata.normalize(self.normalization_type, text)
 
 
+class Lowercase(Normalizer):
+    """Lowercase Normalizer Class."""
+
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        return text.lower()
+
+
+class RegexNormalizer(Normalizer):
+    """Normalization class that substitutes regex matches with either an empty string or a
+    replacement string."""
+
+    def __init__(self, regex_norm_rules):
+        """Creates a Regex Normalizer instance.
+
+        Args:
+            regex_norm_rules (List[Dict]): List of regex normalization rules represented as
+                dictionaries. The example rule below removes any text in parentheses.
+                {
+                    "description": "remove_text_in_parantheses",
+                    "pattern": "\(.+?\)",
+                    "replacement": ""
+                }
+        """
+        super().__init__()
+        self.regex_norm_rules = regex_norm_rules
+
+    def normalize(self, text):
+        """Apply Regex substitutions to the given text.
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        # TODO: Handle case if the "pattern" or "replacement" is none
```

Comment:
For the to-do:
Maybe consider falling back to the ``MindMeldRegexNormalizer`` and throwing a warning?

Or language subset specific defaults (as discussed).

Reply:
For Regex language-specific defaults, the plan is to split Each Regex rule into its own class (as Travis suggested). Doing so, we would no longer have `MindMeldRegexNormalizer`. This would make it easier to maintain language specific defaults. Here is an example: 

```
{ 
      “normalizers”: [
              “RemoveUnderscores”,
              “SeparateAposSChars”,
              “RemoveSpecialCharsBetweenWords”
      ]
}
```

While it is a bit more verbose, this approach is fully customizable and is what HuggingFace follows as well. As Vijay mentioned, for certain languages such as Arabic we would explicitly require developers to use their own Regex rules. As it is not immediately clear which languages would use which Regex norm rules.